SimCalorimetry/HcalSimAlgos/src/HcalSiPM.cc

```diff
@@ -68,13 +103,12 @@ const HcalSiPM::cdfpair& HcalSiPM::BorelCDF(unsigned int k) {
 
     // calculate cdf[i]
     for (++i;; ++i) {
-      b = Borel(k + i, theCrossTalk, k);
-      sumb += b;
+      iFact *= double(i);
+      sumb += Borel(i, theCrossTalk, k, iFact);
       cdf.push_back(sumb);
-      if (1 - sumb < EPSILON)
+      if (1. - sumb < EPSILON)
         break;
     }
```

Comment:
I think wfs `11630.0` and `11723.17` get actually stuck by this change (can be checked locally). `iFact` becomes `inf` when `i >= 171`, `Borel` returns `0`, and the `break` never hits.

Reply:
ok. so I suggest to limit the loop to 170